src/nuclio/common/components/edit-item/edit-item.component.js

```diff
@@ -100,6 +107,47 @@
                         };
                     })
                     .value();
+
+                ctrl.annotations = lodash.chain(ctrl.item.annotations)
+                    .defaultTo([])
+                    .map(function (value, key) {
+                        return {
+                            name: key,
+                            value: value,
+                            ui: {
+                                editModeActive: false,
+                                isFormValid: true,
+                                name: 'trigger.annotation'
+                            }
+                        };
+                    })
+                    .value();
+            }
+
+            if (isKafkaTrigger()) {
+                if (lodash.isNil(ctrl.item.attributes.initialOffset)) {
+                    ctrl.item.attributes.initialOffset = 'latest';
+                }
+
+                if (lodash.isNil(ctrl.item.attributes.sasl)) {
+                    ctrl.item.attributes.sasl = {};
+                }
+
+                ctrl.item.attributes.sasl = lodash.merge({
+                    enable: false,
+                    user: '',
+                    password: ''
+                }, ctrl.item.attributes.sasl);
```

Comment:
Did you mean:
```js
lodash.merge(ctrl.item.attributes.sasl, {
    enable: false,
    user: '',
    password: ''
}, ctrl.item.attributes.sasl);
```
?

It is not the same thing.
What you wrote will always ignore the current value of `ctrl.item.attributes.sasl` and assign the result value of calling `lodash.merge(...)` as its new value (it's okay, if this was the desired behavior).
What I wrote will not assign a new object to `ctrl.item.attributes.sasl` but will only override specific properties by the `merge` method.